internal/db/sqltest/tests/session/session_state_valid_transition_test.sql

```diff
@@ -0,0 +1,25 @@
+begin;
+    select plan(6);
+
+    -- Ensure session state table is populated
+    select is(count(*), 1::bigint) from session_state where session_id = 's1_____clare' and state='pending';
+    select is(count(*), 1::bigint) from session_state where session_id = 's1_____carly' and state='pending';
+
+    -- Valid state transition from pending to active
+    insert into session_state
+    ( session_id, state)
+    values
+        ('s1_____clare','active');
+
+    select is(count(*), 2::bigint) from session_state where session_id = 's1_____clare';
+    select is(count(*), 1::bigint) from session_state where session_id = 's1_____clare' and state='active';
+
+    -- Valid state transition from active to terminated
+    insert into session_state
+    ( session_id, state)
+    values
+        ('s1_____carly','terminated');
+    select is(count(*), 2::bigint) from session_state where session_id = 's1_____carly';
+    select is(count(*), 1::bigint) from session_state where session_id = 's1_____carly' and state='terminated';
```

Comment:
based on the assertion on line 6, it looks like `s1_____carly` is in the `pending` state, so this is checking the transition from pending to terminated.